clients/python/moondream/cloud_vl.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+import urllib.request
+import base64
+from PIL import Image
+from io import BytesIO
+from typing import Union, Optional, Literal
+
+
+from .types import (
+    VLM,
+    CaptionOutput,
+    EncodedImage,
+    QueryOutput,
+    DetectOutput,
+    SamplingSettings,
+)
+
+
+class CloudVL(VLM):
+    @classmethod
+    def from_api_key(
+        cls, api_key: str, api_url: str = None, version: str = "v1"
+    ) -> "CloudVL":
+        """Initialize a CloudVL instance with an API key.
+
+        Args:
+            api_key (str): The API key for authentication
+            api_url (str, optional): Custom API URL. Defaults to None.
+            version (str, optional): API version. Defaults to "v1".
+
+        Returns:
+            CloudVL: An initialized CloudVL instance
+        """
+        return cls(api_key=api_key, api_url=api_url, version=version)
+
+    def __init__(self, api_key: str = None, api_url: str = None, version: str = "v1"):
+        self.api_key = api_key or os.getenv("MOONDREAM_API_KEY")
+        if not self.api_key:
+            raise ValueError(
+                "API key is required. Set MOONDREAM_API_KEY env var or pass api_key"
+            )
+
+        self.api_url = api_url or f"https://api.moondream.ai/{version}"
+        self.client = urllib.request.urlopen
+        self.headers = {"X-Moondream-Auth": self.api_key}
+
+    def _encode_image(self, image: Image.Image) -> str:
+        try:
+            if image.mode != "RGBA":
+                image = image.convert("RGBA")
+            buffered = BytesIO()
+            image.save(buffered, format="PNG")
+            img_str = base64.b64encode(buffered.getvalue()).decode()
+            return f"data:image/png;base64,{img_str}"
+        except Exception as e:
+            raise ValueError(f"Failed to convert image to PNG: {str(e)}")
+
+    def encode_image(self, image: Union[Image.Image, EncodedImage]) -> EncodedImage:
+        raise NotImplementedError("encode_image is not supported for cloud inference")
+
+    def _stream_response(self, req):
+        """Helper function to stream response chunks from the API."""
+        with urllib.request.urlopen(req) as response:
+            for line in response:
+                if not line:
+                    continue
+                line = line.decode("utf-8")
+                if line.startswith("data: "):
+                    try:
+                        chunk = json.loads(line[6:])
+                        if chunk.get("completed"):
+                            break
+                        yield chunk.get("chunk", "")
```

Comment:
Seems better to write this as the following?

```python
if "chunk" in chunk:
    yield chunk["chunk"]
```

That way we wouldn't yield empty strings. Also we should come up with a better variable name than chunk, `chunk["chunk"]` is a little weird.